File: 2022/go/day02.go
package main

/*
Day 02: Rock Paper Scissors

https://adventofcode.com/2022/day/2
*/

import (
	"flag"
	"fmt"
	"io/ioutil"
	"strings"
)

var inputFile = flag.String("input", "../inputs/day02.txt", "Relative path to input-file")

func star1(data []string) {
	res := 0
	for _, line := range data {
		rs := strings.Split(line, " ")
		if rs[0] == "A" { // Rock
			if rs[1] == "X" { // Rock
				res += 1
				res += 3
			} else if rs[1] == "Y" { // Paper
				res += 2
				res += 6
			} else if rs[1] == "Z" { // Siccors
				res += 3
				res += 0
			}
		} else if rs[0] == "B" { // Paper
			if rs[1] == "X" { // Rock
				res += 1
				res += 0
			} else if rs[1] == "Y" { // Paper
				res += 2
				res += 3
			} else if rs[1] == "Z" { // Siccors
				res += 3
				res += 6
			}
		} else if rs[0] == "C" { // Siccors
			if rs[1] == "X" { // Rock
				res += 1
				res += 6
			} else if rs[1] == "Y" { // Paper
				res += 2
				res += 0
			} else if rs[1] == "Z" { // Siccors
				res += 3
				res += 3
			}
		}
	}
	fmt.Println("Star 1:", res)
}

func star2(data []string) {
	res := 0
	for _, line := range data {
		rs := strings.Split(line, " ")
		if rs[0] == "A" { // Rock
			if rs[1] == "X" { // Lose
				res += 3
				res += 0
			} else if rs[1] == "Y" { // Draw
				res += 1
				res += 3
			} else if rs[1] == "Z" { // Win
				res += 2
				res += 6
			}
		} else if rs[0] == "B" { // Paper
			if rs[1] == "X" { // Lose
				res += 1
				res += 0
			} else if rs[1] == "Y" { // Draw
				res += 2
				res += 3
			} else if rs[1] == "Z" { // Win
				res += 3
				res += 6
			}
		} else if rs[0] == "C" { // Siccors
			if rs[1] == "X" { // Lose
				res += 2
				res += 0
			} else if rs[1] == "Y" { // Draw
				res += 3
				res += 3
			} else if rs[1] == "Z" { // Win
				res += 1
				res += 6
			}
		}
	}
	fmt.Println("Star 2:", res)
}

func main() {
	flag.Parse()
	raw, err := ioutil.ReadFile(*inputFile)
	if err != nil {
		panic("Invalid file")
	}
	rawStr := strings.TrimSpace(string(raw))
	data := strings.Split(rawStr, "\n")
	star1(data)
	star2(data)
}

File: 2024/day02/day02.py

"""
Day 02: https://adventofcode.com/2024/day/2
"""
       
def checkRules(l, max=3):
    prev = 0
    incr = False
    for i,v in enumerate(l):
        if i == 0:
            prev = v
            continue
        elif i == 1:
            if (prev == v) or abs(prev-v)>max:
                return False
            elif (prev < v):
                incr = True
            else:
                incr = False
        elif(incr):
            if (prev >= v) or abs(prev-v)>max:
                return False
        else:
            if (prev <= v) or abs(prev-v)>max:
                return False
        prev = v
    return True
        

def star1(file):
    N = 0
    for line in file:
        values = line.strip('\n').split(' ')
        l = [int(s) for s in values]
        if checkRules(l):
            N += 1
    return N

def star2(file):
    return file

def __main__():
    day2input = open("/repo/AdventOfCode/2024/inputs/day02.txt", 'r')
    ans1 = star1(day2input)
    ans2 = star2(day2input)
    print("Star 1:", ans1)
    print("Star 2:", ans2)

__main__()